perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -2159,13 +2631,20 @@ public Attribute getAttributeById(PerunSession sess, Group group, int id) throws
 
 	@Override
 	public Attribute getAttributeById(PerunSession sess, Host host, int id) throws InternalErrorException, AttributeNotExistsException {
+		Attribute attributesCache = null;
+
 		if(!CacheManager.isCacheDisabled()) {
 			Attribute attr = perun.getCacheManager().getAttributeById(id, new Holder(host.getId(), Holder.HolderType.HOST), null);
-			return setValueOfAttribute(sess, attr, host, null);
+			attributesCache = setValueOfAttribute(sess, attr, host, null);
 		}
 
 		try {
-			return jdbc.queryForObject("select " + getAttributeMappingSelectQuery("host_attr_values") + " from attr_names left join host_attr_values on id=attr_id and host_id=? where id=?", new SingleBeanAttributeRowMapper<>(sess, this, host), host.getId(), id);
+			Attribute attributesDB = jdbc.queryForObject("select " + getAttributeMappingSelectQuery("host_attr_values") + " from attr_names left join host_attr_values on id=attr_id and host_id=? where id=?", new SingleBeanAttributeRowMapper<>(sess, this, host), host.getId(), id);
+
+			String message = "The attributes do not match, attribute from cache is: " + attributesCache + " while attribute from db is: " + attributesDB + " the attribute belongs to " + host + " and its id is " + id;
```

Comment:
You probably doesn't have to specify ID of attribute in the log message, because that is the identifier, by which we retrieve it from the cache/DB. Then if results are not the same, at lease one must not be empty and hence logged attributes will contain attribute id.